Mancala.hs

```diff
@@ -152,8 +223,27 @@ capture_pieces end_index (my_old_side, other_old_side) (my_side, other_side)
 -- ([4,4,4,4,0,5,1],[5,5,4,4,4,4,0])
 
 
---update_available_actions :: Board -> ([Action], [Action]) 
---win :: Board -> Bool
+--update_available_actions :: Board -> ([Action], [Action])
+
+win :: Action -> Board -> Bool
+win (Action n) board 
+  | oneSideClear boardAfter = myStore > oppStore
+  | otherwise  = False 
+  where
+   boardAfter = move_pieces (Action n) board
```

Comment:
This should probably goes into the Game type. I imagine win would just take the updated board (after moving pieces, after doing any captures)

Reply:
The example from Poole's file took an action as well, so I thought I'd try to replicate that as much as possible. I can also modify this to just take in a board (it's pretty much the same thing i guess.) This implementation's also missing a check for capture, but we can easily do that once we get move to return an index.